runtime/src/snapshot_utils.rs

```diff
@@ -1780,6 +1864,106 @@ where
     })
 }
 
+/// Streams snapshot dir files across channel
+/// Follow the flow of streaming_unarchive_snapshot(), but handle the from_dir case.
+fn streaming_snapshot_dir_files(
+    file_sender: Sender<PathBuf>,
+    snapshot_file_path: impl Into<PathBuf>,
+    snapshot_version_path: impl Into<PathBuf>,
+    account_paths: &[PathBuf],
+) {
+    file_sender
+        .send(snapshot_file_path.into())
+        .expect("send snapshot file path");
+    file_sender
+        .send(snapshot_version_path.into())
+        .expect("send snapshot version path");
+
+    for account_path in account_paths {
+        for file in fs::read_dir(account_path).expect("read account path dir") {
+            let file_path = file.unwrap().path().to_path_buf();
+            file_sender.send(file_path).expect("send account file path");
+        }
+    }
+}
+
+/// Perform the common tasks when deserialize a snapshot.  Handles reading snapshot file, reading the version file,
+/// and then returning those fields plus the rebuilt storage
+fn build_storage_from_snapshot_dir(
+    snapshot_info: &BankSnapshotInfo,
+    account_paths: &[PathBuf],
+    next_append_vec_id: Arc<AtomicAppendVecId>,
+) -> Result<AccountStorageMap> {
```

Comment:
This introduced several ways to panic, but should be returning a result. Let's catch those `expects` by returning an `Err` instead.

Reply:
removed expect.

Reply:
the existing examples all use unwrap for send().

Reply:
done.